### PR TITLE
Remove nodeport from headless service

### LINKED ISSUE #8763: stable/rabbitmq install error Forbidden: may not be used when type is 'ClusterIP'
**Is this a request for help?**: Yes
**BUG REPORT**

**Version of Helm and Kubernetes**:
Kubernetes 1.10

**Which chart**:
stable/rabbitmq

**What happened**:

Get this error message: 

> Error: release my-rabbit failed: Service "my-rabbit-rabbitmq-headless" is invalid: spec.ports[1].nodePort: Forbidden: may not be used when `type` is 'ClusterIP'

**What you expected to happen**:

The chart used to install, but some recent changes to stable/rabbitmq has broken this.

**How to reproduce it** (as minimally and precisely as possible):

Run example here: https://www.katacoda.com/javajon/courses/kubernetes-applications/rabbitmq

or 

helm install stable/rabbitmq \
--name my-rabbit \
--namespace rabbit \
--set replicas=3 \
--set serviceType=NodePort \
--set rabbitmq.managerPort=31111 \
--set rabbitmq.nodePort=31112 \
--set rabbitmq.username=admin \
--set rabbitmq.password=admin \
--set rbacEnabled=false

**Anything else we need to know**:

Consider the problem is [svc-headless.yaml](https://github.com/helm/charts/blob/master/stable/rabbitmq/templates/svc-headless.yaml) allows the application of nodePort to 

  - name: amqp
    port: {{ .Values.rabbitmq.amqpPort }}
    targetPort: amqp
    {{- if (and (eq .Values.serviceType "NodePort") (not (empty .Values.rabbitmq.nodePort))) }}
    nodePort: {{ .Values.rabbitmq.nodePort }}
    {{- end }}

yet type at top of yaml is fixed to "clusterIP: None"

### PR DESCRIPTION
**Pull request by dene14**
- fixes #8763

#### Special notes for your reviewer:
Fixing crap I've introduced earlier. Headless service shouldn't contain any modes other than ClusterIP.

@carrodher @desaintmartin @juan131 @prydonius @sameersbn @tompizmor

### CODE REVIEW

File: stable/rabbitmq/templates/svc-headless.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ metadata:
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
 spec:
+  type: ClusterIP
```
Comment (juan131) [inline stable/rabbitmq/templates/svc-headless.yaml]: That's the default value. It's not necessary to specify it.

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: stable/rabbitmq/templates/svc-headless.yaml (hunk 1 above)
Comment (dene14) [reply stable/rabbitmq/templates/svc-headless.yaml]: I though it's a best practice to have everything defined. Shall I remove?

Comment (dene14) [reply stable/rabbitmq/templates/svc-headless.yaml]: @juan131

Comment (juan131) [reply stable/rabbitmq/templates/svc-headless.yaml]: If you'd rather define it, I'd make it configurable through **values.yaml**

**Comment by carrodher** [comment]
It seems a new chart version appears, can you rebase the changes and bump again the version if needed?

**Comment by dene14** [comment]
@juan131 @carrodher Ready for merge.

**Pull request merged**